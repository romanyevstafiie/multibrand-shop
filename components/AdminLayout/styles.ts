import styled from 'styled-components';

export const LayoutContainer = styled.section`
  display: grid;
  grid-template-columns: 1fr 8fr;
  min-height: 100vh;
`;

export const ChildrenWrapper = styled.main`
  background-color: ${({ theme }) => theme.dashboardGrayBg};
`;
