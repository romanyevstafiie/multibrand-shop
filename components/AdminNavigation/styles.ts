import styled from 'styled-components';

export const StyledAdminNavigation = styled.aside`
  display: flex;
  align-items: center;
  justify-content: center;
  position: sticky;
  top: 0;
  padding: 5rem 1rem;

  li {
    margin: 1rem 0;
  }
`;
