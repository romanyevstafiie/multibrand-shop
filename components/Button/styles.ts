import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;

const hoveredButton = css`
  background: ${({ theme }) => theme.secondaryGolden};
`;

const focusedButton = css`
  box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryGolden};
`;

const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.primaryGolden}
      repeating-linear-gradient(
        60deg,
        transparent,
        transparent 10px,
        #ffa500 10px,
        #ffc130 20px
      );
    animation: ${loadingKeyframes} 1s infinite linear;
  }
  & > span {
    opacity: 0.5;
  }
  cursor: wait;
`;

const completedButton = css`
  padding: 0 2rem;
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

export const StyledButton = styled.button<Props>`
  position: relative;
  overflow: hidden;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 700;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) =>
    theme.mode === 'dark' ? theme.primaryWhite : theme.secondaryBg};
  cursor: pointer;
  outline: none;
  & > span {
    position: relative;
    display: flex;
    align-items: center;
    z-index: 1;
  }
  ${(p) => !p.isLoading && defaultButton};
  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }
  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }
  &:disabled {
    ${disabledButton};
  }
  ${(p) => p.isLoading && loadingButton};
  ${(p) => p.isCompleted && completedButton};
`;
