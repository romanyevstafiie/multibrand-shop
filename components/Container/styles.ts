import styled from 'styled-components';

export const StyledContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: ${({ theme }) => theme.maxWidth};
  margin: 0 auto;
  padding: 0 1rem;

  a {
    text-decoration: none;
  }
`;
