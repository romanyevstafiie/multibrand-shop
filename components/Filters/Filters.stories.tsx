import { Story } from '@storybook/react';
import React from 'react';
import Filters from './Filters';

export default {
  title: 'Utility/Filter/Filters Component',
  component: Filters,
};

const Template: Story = (args) => <Filters {...args} />;
export const Primary = Template.bind({});
Primary.args = {};
