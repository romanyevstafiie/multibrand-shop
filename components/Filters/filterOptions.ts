import { Department, ProductSize, SneakerBrand } from '../../types/index';
export const filterBrands: SneakerBrand[] = [
  'Adidas',
  'Air Jordan',
  'Asics',
  'Diadora',
  'Converse',
  'Native',
  'Yeezy',
  'Supreme',
  'Timberland',
];

export const filterDepartment: Department[] = ['kid', 'men', 'women'];

export const filterSize: ProductSize[] = [
  '1',
  '1.5',
  '2',
  '2.5',
  '3',
  '3.5',
  '4',
  '4.5',
  '5',
  '5.5',
  '6',
  '6.5',
  '7',
  '7.5',
  '8',
  '8.5',
  '9',
  '9.5',
  '10',
  '10.5',
  '11',
  '11.5',
  '12',
  '13',
  '14',
  '15',
  'onesize',
  'MISC',
  's',
  'm',
  'l',
  'xl',
  'xxl',
  'xs',
];
