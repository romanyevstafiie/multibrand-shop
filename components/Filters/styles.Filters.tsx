import styled from 'styled-components';

export const Container = styled.aside`
  position: sticky;
  top: 20%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 20%;
  height: 35%;
  padding: 1.25rem;
  overflow: hidden;
  background: ${({ theme }) => theme.secondaryBg};
  border: 1px solid ${({ theme }) => theme.borderGrayColor};
  border-radius: ${({ theme }) => theme.borderRadius};
`;

export const ClearFilters = styled.button`
  cursor: pointer;
  background: transparent;
  border: 1px solid ${({ theme }) => theme.borderGrayColor};
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) => theme.primaryWhite};
  padding: 0.5rem 1rem;
  margin: 0.25rem;
  text-transform: uppercase;
  font-weight: 700;

  &:hover {
    background-color: ${({ theme }) => theme.primaryGolden};
  }
`;
