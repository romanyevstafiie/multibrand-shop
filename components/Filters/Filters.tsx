import React, { useCallback } from 'react';
import { useActions } from '../../hooks/useActions';
import { useTypedSelector } from '../../hooks/useTypedSelector';
import CardHeader from '../CardHeader/CardHeader';
import FilterColorsOption from '../FilterColorOption/FilterColorOption';
import FilterTextOption from '../FilterTextOption/FilterTextOption';
import RemoveFilter from '../RemoveFilter/RemoveFilter';
import {
  filterBrands,
  filterColors,
  filterDepartment,
  filterSize,
} from './filterOptions';
import * as Styled from './styles.Filters';

/**
 *This component is responsible for global filtering
 *@function Filters
 *@returns {JSX.Element} - Rendered  Filters component
 */
const Filters = (): JSX.Element => {
  // All filter values we will get from redux
  const { brand, color, department, size } = useTypedSelector(
    (state) => state.products.filters
  );
  // Action creators
  const { addFilterOption, removeAllFilters } = useActions();

  const handleRemoveFilters = () => removeAllFilters();

  // Each callback set filter value and simultaneously dispatch action to add this filter to global filter object
  const setBrand = useCallback(
    (val: string) => {
      addFilterOption({ filterName: 'brand', filterValue: val });
    },
    [addFilterOption]
  );

  const setColor = useCallback(
    (val: string) => {
      addFilterOption({ filterName: 'color', filterValue: val });
    },
    [addFilterOption]
  );

  const setDepartment = useCallback(
    (val: string) => {
      addFilterOption({ filterName: 'department', filterValue: val });
    },
    [addFilterOption]
  );

  const setSize = useCallback(
    (val: string) => {
      addFilterOption({ filterName: 'size', filterValue: val });
    },
    [addFilterOption]
  );

  // will render remove filters button if there is one filter at least
  const removeFilterButton = (): JSX.Element | null => {
    if (brand || color || department || size)
      return (
        <Styled.ClearFilters onClick={handleRemoveFilters}>
          Clear filters
        </Styled.ClearFilters>
      );
    return null;
  };

  return (
    <Styled.Container>
      <CardHeader headerText="Filters" />

      {brand && <RemoveFilter filterName={brand} removeFilter={setBrand} />}
      {color && <RemoveFilter filterName={color} removeFilter={setColor} />}
      {department && (
        <RemoveFilter filterName={department} removeFilter={setDepartment} />
      )}
      {size && <RemoveFilter filterName={size} removeFilter={setSize} />}
      {removeFilterButton()}

      <FilterTextOption
        filterBy="Brand"
        filterOptions={filterBrands}
        setFilter={setBrand}
      />
      <FilterColorsOption
        filterBy="Color"
        filterOptions={filterColors}
        setFilter={setColor}
      />
      <FilterTextOption
        filterBy="Department"
        filterOptions={filterDepartment}
        setFilter={setDepartment}
      />
      <FilterTextOption
        filterBy="Size"
        filterOptions={filterSize}
        setFilter={setSize}
      />
    </Styled.Container>
  );
};

export default Filters;
