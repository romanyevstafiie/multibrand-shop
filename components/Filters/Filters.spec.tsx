import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { mainTheme } from '../../styles/mainTheme';
import { initialReduxStore } from '../../test/initialReduxStore';
import { storeFactory } from '../../test/testUtils';
import {
  default as FilterColorsOption,
  default as FilterTextOption,
} from '../FilterColorOption/FilterColorOption';
import Filters from './Filters';

/**
 * Setup function for the component
 * @returns {ShallowWrapper}
 */
const setup = () => {
  const store = storeFactory(initialReduxStore);
  return mount(
    <ThemeProvider theme={mainTheme}>
      <Provider store={store}>
        <Filters />
      </Provider>
    </ThemeProvider>
  );
};

describe('All Filters Component', () => {
  const wrap = setup();

  it('should render without throwing an error', () => {
    expect(wrap);
    expect(wrap.length).toBe(1);
  });
  it('should render all text filters ', () => {
    const textFilters = wrap.find(FilterTextOption);

    expect(textFilters.length).toBe(1);
  });
  it('should render color filter ', () => {
    const colorFilter = wrap.find(FilterColorsOption);

    expect(colorFilter.length).toBe(1);
  });
});
