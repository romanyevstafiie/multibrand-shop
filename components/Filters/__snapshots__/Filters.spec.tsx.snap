// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Filters Component SNAPSHOT it applies default styles 1`] = `
<aside
  className="stylesFilters__Container-sc-1wcbpf8-0 hClXvr"
>
  <h3
    className="CardHeader__StyledHeader-sc-fpyt0y-0 ezAfQK"
  >
    Filters
  </h3>
  
  
  
  
  <div
    className="stylesFilterTextOption__Container-sc-9ak2es-0 jeogpl"
  >
    <details
      className="stylesFilterTextOption__FilterOption-sc-9ak2es-1"
    >
      <summary>
        Brand
      </summary>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Adidas
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Air Jordan
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Asics
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Diadora
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Converse
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Native
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Yeezy
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Supreme
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        Timberland
      </button>
    </details>
  </div>
  <div
    className="stylesFilterColorOptions__Container-sc-ka1lzr-0 bXWAim"
  >
    <details
      className="stylesFilterColorOptions__FilterOption-sc-ka1lzr-1"
    >
      <summary>
        Color
      </summary>
      <div
        className="stylesFilterColorOptions__ColorsGrid-sc-ka1lzr-2 gPvYhI"
      >
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 hCLZtM"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 iIkYjf"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 jEaXJu"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 eRZsnu"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 jszyzk"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 UZPjb"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 dNdEuh"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 gpCxmL"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 VhVRP"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 djgifk"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 gXNoJM"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 migfm"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 bPmVSn"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 bnYwBn"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 bqLyMH"
          onClick={[Function]}
        />
        <button
          className="stylesFilterColorOptions__FilterButton-sc-ka1lzr-3 epQCIs"
          onClick={[Function]}
        />
      </div>
    </details>
  </div>
  <div
    className="stylesFilterTextOption__Container-sc-9ak2es-0 jeogpl"
  >
    <details
      className="stylesFilterTextOption__FilterOption-sc-9ak2es-1"
    >
      <summary>
        Department
      </summary>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        kid
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        men
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        women
      </button>
    </details>
  </div>
  <div
    className="stylesFilterTextOption__Container-sc-9ak2es-0 jeogpl"
  >
    <details
      className="stylesFilterTextOption__FilterOption-sc-9ak2es-1"
    >
      <summary>
        Size
      </summary>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        1.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        2.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        3.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        4.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        5.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        6.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        7.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        8.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        9.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        10
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        10.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        11
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        11.5
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        12
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        13
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        14
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        15
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        onesize
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        MISC
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        s
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        m
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        l
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        xl
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        xxl
      </button>
      <button
        className="stylesFilterTextOption__FilterButton-sc-9ak2es-2 lihCeO"
        onClick={[Function]}
      >
        xs
      </button>
    </details>
  </div>
</aside>
`;
